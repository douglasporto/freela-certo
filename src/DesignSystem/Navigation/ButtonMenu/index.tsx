export { default } from './ButtonMenu'
export * from './ButtonMenu'
