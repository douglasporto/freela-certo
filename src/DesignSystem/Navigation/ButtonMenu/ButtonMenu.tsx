import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>

export type ButtonMenuProps = {
  children: React.ReactNode
  isActive?: boolean
  className?: string
  icon?: JSX.Element
  href: string
} & ButtonTypes

export default function ButtonMenu({
  children,
  className,
  isActive = false,
  href,
  icon
}: ButtonMenuProps) {
  return (
    <S.Wrapper href={href} className={className} isActive={isActive}>
      {icon}
      {children}
    </S.Wrapper>
  )
}
