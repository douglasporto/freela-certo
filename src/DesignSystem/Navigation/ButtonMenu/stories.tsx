import { Story, Meta } from '@storybook/react/types-6-0'

import ButtonMenu, { ButtonMenuProps } from '.'

export default {
  title: 'Navigation/ButtonMenu',
  component: ButtonMenu,
  argTypes: {
    href: {
      type: null
    }
  },
  args: {
    children: 'Dashboard',
    href: '',
    isActive: false
  }
} as Meta

export const Default: Story<ButtonMenuProps> = (args) => (
  <div style={{ width: '300px', margin: 'auto' }}>
    <ButtonMenu {...args} />
  </div>
)
