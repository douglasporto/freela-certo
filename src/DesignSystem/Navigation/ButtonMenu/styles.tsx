import styled, { css, DefaultTheme } from 'styled-components'

import { Button } from '@material-ui/core'

import { ButtonMenuProps } from './ButtonMenu'

const wrapperModifiers = {
  isActive: (theme: DefaultTheme) => css`
    background-color: ${theme.colors.background.second} !important;
    color: ${theme.colors.primary} !important;
  `
}

export const Wrapper = styled(({ children, className }) => (
  <Button variant="text" className={className}>
    {children}
  </Button>
))<Pick<ButtonMenuProps, 'isActive'>>`
  ${({ theme, isActive }) => css`
    width: 100%;
    background-color: initial;
    color: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    border: 0;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    font-size: ${theme.font.sizes.small}rem;
    font-weight: ${theme.font.bold};
    cursor: pointer;
    transition: ${theme.transition.default};
    :hover {
      background-color: ${theme.colors.background.second};
      color: ${theme.colors.primary};
    }
    ${isActive && wrapperModifiers.isActive(theme)}
  `}
`
