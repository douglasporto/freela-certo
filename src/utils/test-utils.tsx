import { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles'

import muiTheme from 'styles/muiTheme'
import theme from 'styles/theme'

const customRender = (ui: ReactElement) =>
  render(
    <MuiThemeProvider theme={muiTheme}>
      <ThemeProvider theme={{ ...muiTheme, ...theme } as const}>
        {ui}
      </ThemeProvider>
    </MuiThemeProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
