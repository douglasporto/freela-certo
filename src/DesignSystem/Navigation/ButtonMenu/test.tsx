import { render, screen } from 'utils/test-utils'

import ButtonMenu from '.'

describe('<ButtonMenu />', () => {
  it('should render the button menu', () => {
    render(<ButtonMenu href="">Dashboard</ButtonMenu>)

    expect(screen.getByText(/Dashboard/i)).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Dashboard/i })).toHaveStyle({
      width: '100%',
      backgroundColor: 'initial'
    })
  })

  it('should render active the button menu', () => {
    const { container } = render(
      <ButtonMenu href="" isActive>
        Dashboard
      </ButtonMenu>
    )

    expect(screen.getByText(/Dashboard/i)).toBeInTheDocument()

    expect(screen.getByText(/Dashboard/i)).toHaveStyle({
      width: '100%',
      backgroundColor: '#1A223F !important',
      color: '#7C4DFF !important'
    })

    expect(container).toMatchSnapshot()
  })
})
