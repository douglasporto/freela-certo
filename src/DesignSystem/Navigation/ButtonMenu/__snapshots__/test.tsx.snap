// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonMenu /> should render active the button menu 1`] = `
.c0 {
  width: 100%;
  background-color: initial;
  color: #D7DCEC;
  border-radius: 0.4rem;
  border: 0;
  padding: 0.75rem 1rem;
  font-size: 0.875rem;
  font-weight: 600;
  cursor: pointer;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  background-color: #1A223F !important;
  color: #7C4DFF !important;
}

.c0:hover {
  background-color: #1A223F;
  color: #7C4DFF;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text c0"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Dashboard
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`
